enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/analytics.ts

```diff
@@ -5,66 +5,74 @@ import type {
+ */
+const EMBED_OPTIONS_TO_TRACK = [
+  "drills",
+  "withTitle",
+  "withDownloads",
+  "isSaveEnabled",
+  "readOnly",
+] as const satisfies SdkIframeEmbedSettingKey[];
+
+export const trackEmbedWizardOpened = () =>
+  trackSimpleEvent({ event: "embed_wizard_opened" });
+
+export const trackEmbedWizardExperienceCompleted = (
   experience: SdkIframeEmbedSetupExperience,
 ) =>
   trackSimpleEvent({
-    event: "embed_wizard_experience_selected",
+    event: "embed_wizard_experience_completed",
     event_detail: experience,
   });
 
-export const trackEmbedWizardResourceSelected = (
-  resourceId: number,
+export const trackEmbedWizardResourceSelectionCompleted = (
   experience: SdkIframeEmbedSetupExperience,
 ) =>
   trackSimpleEvent({
-    event: "embed_wizard_resource_selected",
-    target_id: resourceId,
+    event: "embed_wizard_resource_selection_completed",
     event_detail: experience,
   });
 
-export const trackEmbedWizardCodeCopied = () =>
-  trackSimpleEvent({
-    event: "embed_wizard_code_copied",
-  });
-
-export const trackEmbedWizardSettingsUpdated = (
+export const trackEmbedWizardOptionsCompleted = (
   settings: Partial<SdkIframeEmbedSettings>,
 ) => {
-  const filteredSettings = filterEmptySettings(settings);
+  const options: Record<string, string> = {
+    theme: settings.theme?.colors ? "custom" : "default",
+    auth: settings.useExistingUserSession ? "user_session" : "sso",
+  };
 
-  const settingKeys = Object.keys(
-    filteredSettings,
-  ) as SdkIframeEmbedSettingKey[];
-
-  const untrackedKeys: SdkIframeEmbedSettingKey[] = [
-    // These keys are not tracked as user settings.
-    "instanceUrl",
+  for (const optionKey of EMBED_OPTIONS_TO_TRACK) {
+    if (!(optionKey in settings)) {
+      continue;
+    }
 
-    // These are tracked via trackEmbedWizardExperienceSelected and trackEmbedWizardResourceSelected
-    "questionId",
-    "dashboardId",
-  ];
+    const value = (settings as any)[optionKey];
 
-  settingKeys.forEach((settingKey) => {
-    if (untrackedKeys.includes(settingKey)) {
-      return;
+    if (value === null || value === undefined) {
+      continue;
     }
 
-    if (settingKey === "useExistingUserSession") {
-      trackSimpleEvent({
-        event: "embed_wizard_auth_selected",
-        event_detail: settings.useExistingUserSession ? "user-session" : "sso",
-      });
+    options[optionKey] = value.toString();
+  }
 
-      return;
-    }
+  const eventDetail = Object.entries(options)
+    .map(([key, value]) => `${key}=${value}`)
+    .join(",");
 
-    trackSimpleEvent({
-      event: "embed_wizard_option_changed",
-      event_detail: settingKey,
-    });
+  trackSimpleEvent({
+    event: "embed_wizard_options_completed",
+    event_detail: eventDetail,
   });
 };
+
+export const trackEmbedWizardCodeCopied = (
+  authMethod: "sso" | "user_session",
+) =>
+  trackSimpleEvent({
+    event: "embed_wizard_code_copied",
+    event_detail: `auth=${authMethod}`,
```

Comment:
The last time I look up how we use Simple events, I see we send enum to `event_detail` in the Simple event. You could try to see if that's still the case. 

I'g suggest we don't do this and just send the authMethod directly. This would make querying it in stats easier. Otherwise, we'll need to filter like `event_detail="auth=sso"` which is a little weird.

See how to query [Simple events here.](https://stats.metabase.com/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7ImRhdGFiYXNlIjo0OCwicXVlcnkiOnsic291cmNlLXRhYmxlIjo1MjMxM30sInR5cGUiOiJxdWVyeSJ9LCJkaXNwbGF5IjoidGFibGUiLCJ2aXN1YWxpemF0aW9uX3NldHRpbmdzIjp7fX0=)

Reply:
@WiNloSt That's fair, let me change this. I was thinking of sending more things in `event_detail` but now that I think about it, it's better to leave that job to `embed_wizard_options_completed` and code_copied should only send the auth.